### PR TITLE
api user-access: Log on and off machine

### PR DESCRIPTION
**Pull request by jstar1**
adds logging for events when a user logs on and
off to a machine

### CODE REVIEW

File: routes/api.js [hunk 1]
```diff
@@ -190,14 +190,16 @@ router.post('/user-access', function(req, res, next) {
...
-      stationLog.addUser(res.locals.obj.userInCache['user'], res.locals.obj.userInCache['badge'], req.headers['station-id']);
-      res.set({
-        'Content-Type': 'text/plain',
-        'station-state': 'enabled',
-        'User-ID-String': res.locals.obj.userInCache['user']
-      })
-      return res.sendStatus(200);
+      db.logEvent('Granted permission to log into station', + req.headers['station-id'] + 'by' + res.locals.obj.user + 'for' + res.locals.obj.userInCache['badge']).then(function(){
+        cache.delete(req.headers['station-id']);
+        stationLog.addUser(res.locals.obj.userInCache['user'], res.locals.obj.userInCache['badge'], req.headers['station-id']);
```
Comment (bsm5) [inline routes/api.js]: 732	Granted permission to log into station	6byY,Xfor111	2018-03-01 20:18:48 +0000

This is what is recorded in the DB log after hitting this log event.  It doesn't look like res.locals.obj.user is working as you think it should be.  Also as you can tell by the 6byY,Xfor111 you need to edit this log to include proper spacing between words.  It should look something like this: 6 by X,Y, for 111

File: routes/api.js [hunk 2]
```diff
@@ -231,11 +233,13 @@ router.post('/user-access', function(req, res, next) {
     };
     if(anyOneOnMachine.user === null) {
       stationLog.addUser(res.locals.obj.user, res.locals.obj.badge, req.headers['station-id']);
+      db.logEvent('log on station', + req.headers['stationd-id'] + 'by user' + res.locals.obj.user).then(function(){
```
Comment (bsm5) [inline routes/api.js]: 734 log on station NaNby userB1,A1

Like I stated above this log also needs to be edited to include proper spacing between words.  It also looks like the req.headers['stationd-id'] and res.locals.obj.user are not working as you had assumed they would.

File: routes/api.js [hunk 3]
```diff
@@ -255,6 +259,8 @@ router.post('/user-access', function(req, res, next) {
         'Content-Type': 'text/plain',
         'station-state': 'disabled',
       });
+      db.logEvent('log off station', + req.headers['stationd-id'] + 'by user' + res.locals.obj.user).then(function(){
```
Comment (bsm5) [inline routes/api.js]: Same as the log on logEvent.  This log needs to be edited to include proper spacing and the same issues with req.headers['stationd-id'] and res.locals.obj.user seem to not be working as you had assumed.

**Comment by jstar1** [comment]
Alright @bsm5, everything should look good now.

**Comment by jstar1** [comment]
Alright @bsm5 , thanks for doing the code review. I think this should do it.

**Pull request merged**